pkgs/applications/science/logic/z3/default.nix

```
@@ -49,9 +49,14 @@ stdenv.mkDerivation rec {
     mkdir -p $python/lib
     mv $lib/lib/python* $python/lib/
     ln -sf $lib/lib/libz3${stdenv.hostPlatform.extensions.sharedLibrary} $python/${python.sitePackages}/z3/lib/libz3${stdenv.hostPlatform.extensions.sharedLibrary}
+  '' + optionalString javaBindings ''
+    mkdir -p $java
+    mv com.microsoft.z3.jar $java
+    mv $lib/lib/libz3java.${stdenv.hostPlatform.extensions.sharedLibrary} $java
```

This is the preferred way to move files to another output:

```suggestion
    moveToOutput com.microsoft.z3.jar $java
    moveToOutput lib/libz3java.${stdenv.hostPlatform.extensions.sharedLibrary} $java
```

Are you sure the JAR file should end up in the root of the output?

---

Thanks for introducing me to `moveToOutput`.

I did some more searching, and found [`addPkgToClassPath`](https://github.com/NixOS/nixpkgs/blob/e7fe577d9803885d1191c6612b95c246cb605dde/pkgs/build-support/setup-hooks/set-java-classpath.sh#L6-L11) which suggests to use `$java/share/java/com.microsoft.z3.jar`.

Fixed in 6f4fe079422.